relayer-cli/src/conclude.rs

```diff
@@ -207,6 +203,24 @@ impl Output {
         Output::with_success().with_msg(msg)
     }
 
+    pub fn combined(outputs: Vec<Self>) -> Output {
```

Comment:
I am not sure if this is the best abstraction for dealing with multiple outputs, since the way this function is implemented, even if a single `Output` has a status `Error` the combined output is created with a status `Error`. 
Another solution would be to create a `fn print(&self)` function that would print an output (to stdout/stderr) and the caller gets to decide how/when to call it.

Reply:
Yeah the `Output` type was initially not meant for displaying the output incrementally, but rather to provide the final output of the command on stdout for consumption by scripts etc. when JSON output is enabled. We currently do incremental output with the `tracing` crate which outputs to stderr. While this is not perfect, I think we should stick with this model for the purpose of this PR and try to revisit this in the future. What do you think?

Reply:
To make it clearer, my suggestion here is to use `tracing` to display status updates incrementally (eg. on each client update), collect the results somewhere in a custom datatype with a manual `Display` instance, and use `Output` on that.

Reply:
It makes a lot of sense to think of the `Output` as the final output of a command. 👍 I have modified the code to remove `Output::combined()` and instead accumulate the results in a newtype which I use to create a final `Output`. 
Here's what the final output looks like -> 
```bash
# two outer errors
.
├─ config error: missing chain for id (07-tendermint-0) in configuration file
└─ config error: missing chain for id (07-tendermint-1) in configuration file

# one inner errors
.
└─.
  └─ chain runtime error: failed while trying to upgrade client id 07-tendermint-0 with error: failed while querying src chain

# one outer for host-0 and two inners errors both for host-1
.
├─ config error: missing chain for id (07-tendermint-0) in configuration file
└─.
  ├─ chain runtime error: failed while trying to upgrade client id 07-tendermint-1 with error: failed while querying src chain
  └─ chain runtime error: failed while trying to upgrade client id 07-tendermint-1 with error: failed while querying src chain

# one outer and 2 inner errors each from different hosts
.
├─ config error: missing chain for id (07-tendermint-0) in configuration file
├─.
│ └─ chain runtime error: failed while trying to upgrade client id 07-tendermint-0 with error: failed while querying src chain
└─.
  └─ chain runtime error: failed while trying to upgrade client id 07-tendermint-0 with error: failed while querying src chain

# one outer error, one inner error and one success message (Vec<IbcEvent> serialized) each from different hosts 
.
├─ config error: missing chain for id (07-tendermint-0) in configuration file
├─.
│ └─ chain runtime error: failed while trying to upgrade client id 07-tendermint-0 with error: failed while querying src chain
└─.
  └─ [
    UpgradeClient(
        UpgradeClient(
            Attributes {
                height: Height {
                    revision: 0,
                    height: 0,
                },
                client_id: ClientId(
                    "07-tendermint-0",
                ),
                client_type: Tendermint,
                consensus_height: Height {
                    revision: 0,
                    height: 0,
                },
            },
        ),
    ),
]

# success message
[[{"UpgradeClient":{"height":{"revision_number":0,"revision_height":0},"client_id":"07-tendermint-0","client_type":"Tendermint","consensus_height":{"revision_number":0,"revision_height":0}}}],[{"UpgradeClient":{"height":{"revision_number":0,"revision_height":0},"client_id":"07-tendermint-0","client_type":"Tendermint","consensus_height":{"revision_number":0,"revision_height":0}}},{"UpgradeClient":{"height":{"revision_number":0,"revision_height":0},"client_id":"07-tendermint-0","client_type":"Tendermint","consensus_height":{"revision_number":0,"revision_height":0}}},{"UpgradeClient":{"height":{"revision_number":0,"revision_height":0},"client_id":"07-tendermint-0","client_type":"Tendermint","consensus_height":{"revision_number":0,"revision_height":0}}},{"UpgradeClient":{"height":{"revision_number":0,"revision_height":0},"client_id":"07-tendermint-0","client_type":"Tendermint","consensus_height":{"revision_number":0,"revision_height":0}}}]]
```
And here's the test used to generate this output -> 
```rust
#[cfg(test)]
mod tests {
    use super::*;
    use ibc::ics02_client::client_type::ClientType;
    use ibc::ics02_client::events::Attributes;
    use ibc_relayer::foreign_client::ForeignClientError;

    fn print_output_buffer(ob: OutputBuffer) {
        match ob.into_result() {
            Ok(events) => println!("{}", serde_json::to_string(&events).unwrap()),
            Err(e) => println!("{}", e),
        }
    }

    #[test]
    fn test_output_buffer() {
        print_output_buffer(OutputBuffer(vec![
            Err(Kind::Config
                .context(format!(
                    "missing chain for id ({}) in configuration file",
                    ClientId::new(ClientType::Tendermint, 0).unwrap()
                ))
                .into()),
            Err(Kind::Config
                .context(format!(
                    "missing chain for id ({}) in configuration file",
                    ClientId::new(ClientType::Tendermint, 1).unwrap()
                ))
                .into()),
        ]));

        print_output_buffer(OutputBuffer(vec![Ok(vec![Err(Kind::Runtime
            .context(ForeignClientError::ClientUpgrade(
                ClientId::new(ClientType::Tendermint, 0).unwrap(),
                "failed while querying src chain".to_owned(),
            ))
            .into())])]));

        print_output_buffer(OutputBuffer(vec![
            Err(Kind::Config
                .context(format!(
                    "missing chain for id ({}) in configuration file",
                    ClientId::new(ClientType::Tendermint, 0).unwrap()
                ))
                .into()),
            Ok(vec![
                Err(Kind::Runtime
                    .context(ForeignClientError::ClientUpgrade(
                        ClientId::new(ClientType::Tendermint, 1).unwrap(),
                        "failed while querying src chain".to_owned(),
                    ))
                    .into()),
                Err(Kind::Runtime
                    .context(ForeignClientError::ClientUpgrade(
                        ClientId::new(ClientType::Tendermint, 1).unwrap(),
                        "failed while querying src chain".to_owned(),
                    ))
                    .into()),
            ]),
        ]));

        print_output_buffer(OutputBuffer(vec![
            Err(Kind::Config
                .context(format!(
                    "missing chain for id ({}) in configuration file",
                    ClientId::new(ClientType::Tendermint, 0).unwrap()
                ))
                .into()),
            Ok(vec![Err(Kind::Runtime
                .context(ForeignClientError::ClientUpgrade(
                    ClientId::new(ClientType::Tendermint, 0).unwrap(),
                    "failed while querying src chain".to_owned(),
                ))
                .into())]),
            Ok(vec![Err(Kind::Runtime
                .context(ForeignClientError::ClientUpgrade(
                    ClientId::new(ClientType::Tendermint, 0).unwrap(),
                    "failed while querying src chain".to_owned(),
                ))
                .into())]),
        ]));

        print_output_buffer(OutputBuffer(vec![
            Err(Kind::Config
                .context(format!(
                    "missing chain for id ({}) in configuration file",
                    ClientId::new(ClientType::Tendermint, 0).unwrap()
                ))
                .into()),
            Ok(vec![Err(Kind::Runtime
                .context(ForeignClientError::ClientUpgrade(
                    ClientId::new(ClientType::Tendermint, 0).unwrap(),
                    "failed while querying src chain".to_owned(),
                ))
                .into())]),
            Ok(vec![Ok(vec![IbcEvent::UpgradeClient(
                Attributes::default().into(),
            )])]),
        ]));

        print_output_buffer(OutputBuffer(vec![
            Ok(vec![Ok(vec![IbcEvent::UpgradeClient(
                Attributes::default().into(),
            )])]),
            Ok(vec![Ok(vec![
                IbcEvent::UpgradeClient(Attributes::default().into()),
                IbcEvent::UpgradeClient(Attributes::default().into()),
                IbcEvent::UpgradeClient(Attributes::default().into()),
                IbcEvent::UpgradeClient(Attributes::default().into()),
            ])]),
        ]));
    }
}
```

Reply:
Looks great! 💯